geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -112,7 +112,26 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val qp = planQuery(filter, iteratorConfig.iterator, output, keyPlanner, cfPlanner)
 
     val table = acc.getSpatioTemporalTable(sft)
-    val iterators = qp.iterators ++ List(Some(stiiIterCfg), densityIterCfg).flatten
+    var iterators: Seq[IteratorSetting] = qp.iterators ++ List(Some(stiiIterCfg), densityIterCfg).flatten
+
+    if(query.getHints.containsKey(QUERY_SIZE_KEY)) {
+      val cfg = new IteratorSetting(iteratorPriority_SimpleFeatureFilteringIterator, "query-size-iterator", classOf[QuerySizeIterator])
```

Comment:
(style) We tend to try and avoid long lines (over 100 characters), so break out the arguments on newlines.  (Look around for examples.)